mon-pix/app/components/module/grain.js

```diff
@@ -22,21 +29,41 @@ export default class ModuleGrain extends Component {
     return this.args.passage.getLastCorrectionForElement(element);
   }
 
+  @action
+  stepperIsFinished() {
+    this.isStepperFinished = true;
+  }
+
   get shouldDisplayContinueButton() {
-    return this.args.canMoveToNextGrain && this.allElementsAreAnswered;
+    if (!this.hasStepper) {
+      return this.args.canMoveToNextGrain && this.allElementsAreAnswered;
+    } else {
+      return this.args.canMoveToNextGrain && this.isStepperFinished && this.allElementsAreAnswered;
+    }
   }
 
   get shouldDisplaySkipButton() {
-    return this.args.canMoveToNextGrain && this.hasAnswerableElements && !this.allElementsAreAnswered;
+    if (!this.hasStepper) {
+      return this.args.canMoveToNextGrain && this.hasAnswerableElements && !this.allElementsAreAnswered;
+    } else {
+      if (!this.isStepperFinished) {
+        return this.args.canMoveToNextGrain;
+      } else {
+        return this.args.canMoveToNextGrain && this.hasAnswerableElements && !this.allElementsAreAnswered;
+      }
+    }
```

Comment:
je me demande si on pourrait pas remplacer ça par :

```suggestion
    if (!this.hasStepper) {
      return this.args.canMoveToNextGrain && this.hasAnswerableElements && !this.allElementsAreAnswered;
    } else if (!this.isStepperFinished) {
        return this.args.canMoveToNextGrain;
    } else {
        return this.args.canMoveToNextGrain && this.hasAnswerableElements && !this.allElementsAreAnswered;
    }
```

Reply:
Revue ensemble avec Dimitri :)